### PR TITLE
adds dependency for Android

### PR DESCRIPTION
**Pull request by SergeyZhukovsky**
closes https://github.com/brave/brave-browser/issues/4887
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/net/BUILD.gn [hunk 1]
```diff
@@ -31,23 +31,15 @@ source_set("net") {
 
   deps = [
     "//base",
+    "//brave/app:brave_generated_resources_grit",
+    "//brave/app/android:common_extensions",
```
Comment (bridiver) [inline browser/net/BUILD.gn]: this has to remain in an `is_android` guard or there will be duplicate symbol errors

Comment (bridiver) [reply browser/net/BUILD.gn]: actually this should be 
```if (is_android) {
  deps += [ "//brave/app/android:common_extensions" ]
} else {
  deps += [ "//extensions/common" ]
}

File: browser/net/BUILD.gn [hunk 2]
```diff
@@ -31,18 +31,15 @@ source_set("net") {
...
     "//extensions/common:common_constants",
     "//net",
   ]
 
-  if (!is_android) {
-    deps += [
-      "//brave/app:brave_generated_resources_grit",
-    ]
-  } else {
+  if (is_android) {
```
Comment (bridiver) [inline browser/net/BUILD.gn]: this is the correct fix for the current problem, but if we need `"//brave/app/android:common_extensions",` then we're also missing a non-android dep for `//extensions/common` so I think we should add it now while we're here :)

Comment (SergeyZhukovsky) [reply browser/net/BUILD.gn]: I pushed the fix for it

**Pull request merged**